File: app/src/main/java/domon/cn/gankio/ui/activity/ImageViewActivity.java
package domon.cn.gankio.ui.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.socks.library.KLog;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import domon.cn.gankio.R;

/**
 * Created by Domon on 16-8-19.
 */
public class ImageViewActivity extends AppCompatActivity {
    @BindView(R.id.myIv)
    ImageView imageView;

    @OnClick(R.id.myIv)
    void onImageViewClice() {
        finish();
    }

    public static void startActivity(Context context, String url) {
        Intent intent = new Intent(context,ImageViewActivity.class);
        intent.putExtra("url", url);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_imageview);
        ButterKnife.bind(this);

        Intent i = getIntent();
        String s = i.getStringExtra("url");
        KLog.e(s);

        Glide.with(this).load(s).into(imageView);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/domon/cn/gankio/ui/activity/ImageViewActivity.kt
package domon.cn.gankio.ui.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.bumptech.glide.Glide
import domon.cn.gankio.R
import kotlinx.android.synthetic.main.activity_imageview.*

/**
 * Created by Domon on 16-8-19.
 */
class ImageViewActivity : AppCompatActivity() {
    lateinit var url: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_imageview)

        url = intent.getStringExtra("url")
        myIv.setOnClickListener { finish() }

        Glide.with(this).load(url).into(myIv)
    }

    // I dont know what is **companion object**
    // answer: kotlin use **companion object** to define static function
    // use **object** to define static class
    companion object {

        fun startActivity(context: Context, url: String) {
            val intent = Intent(context, ImageViewActivity::class.java)
            intent.putExtra("url", url)
            context.startActivity(intent)
        }
    }
}
